plutus-ledger-api/src/Plutus/V1/Ledger/Api.hs

```diff
@@ -224,20 +221,16 @@ mkTermToEvaluate bs args = do
 -- a limit to guard against scripts that run for a long time or loop.
 evaluateScriptRestricting
     :: VerboseMode     -- ^ Whether to produce log output
-    -> CostModelParams -- ^ The cost model to use
+    -> EvaluationContext -- ^ The cost model that should already be synced to the latest ProtocolCostModel
```

Comment:
Is it actually the latest one, or the one that you need for a particular script?  I had the idea that we would need different cost models for scripts created at different times, but that might be wrong, or I could be misunderstanding what's going on here.

Reply:
It is become more tricky with multiple protocol versions indeed, but with the word "latest" I meant the most-recently updated cost model parameters for the particular protocol version. Maybe I should rephrase it.

Reply:
It's also not a cost model: the whole point of having an opaque `EvaluationContext` type is that we can change what's in it silently. So it's just "the evaluation context for this evaluation".